### ISSUE TITLE
Change server IP address

### ISSUE THREAD

**Issue by eletkftbiatorbagy** [phonegap/phonegap-app-desktop#356]
I have two different network interfaces on my mac, and the software is trying to use the one for my iscsi network, which can't be connected by the app.
Is there a way to change the listen IP address ?

**Comment by hermwong** [comment]
Currently the PhoneGap Desktop doesn't allow for changing of the IP address.

For connection related issues you could try the following:

1. make sure the computer with PhoneGap Desktop and the mobile device with PhoneGap Developer App are on the same network
1. make sure the port being used by PhoneGap Desktop's server isn't being blocked by firewall or anti-virus (open that port or use a different port)

**Comment by CtrlC-Root** [comment]
This is an incredibly annoying limitation. On some systems, like OS X, you simply can't disable all the unused network adapters. There should either be a way to set it or it should listen on ```0.0.0.0```.

**Comment by hermwong** [comment]
This is actually a limitation of the `connect-phonegap` module which the `PhoneGap Desktop App` uses to serve projects.

The issue is already logged in the `connect-phonegap` module's issue tracker: https://github.com/phonegap/connect-phonegap/issues/77

**Comment by servolutions** [comment]
I have the same problem. I use VMware on my system and that always adds a couple of virtual network cards to the system. Unfortunately PhoneGap grabs the wrong card, can't use it at all. Bump!

**Comment by hermwong** [comment]
related to: #330

**Comment by Kurlrip** [comment]
Hello, if that could help someone.
I use Windows. I use VMWare too, so i have many card. 
PhoneGap want use an IP from virtual card. So, i "disconnect" (right click on card > disconnect) the 3 virtual card. So only my wifi card is online. Next i restart PhoneGap, and now it use the goog Ip Adress.

**Comment by adriespo** [comment]
I resolved using the tip provided by Kurlrip! Thanks!

**Comment by jarnix** [comment]
Up. Come on... obviously many developers have vmware or other virtualization apps on their computer, therefore multiple network cards. Disconnecting the others is a workaround, not a solution, the ticket should not be closed.

**Comment by alcastle** [comment]
Couldn't even use PhoneGap on OSX, since it was picking a 127.* address for some reason.
Tried uninstalling the app, restarting networking, restarting the system, re-installing; nothing worked.

This hack at least let me work around it for now so I can follow the tutorial. http://stackoverflow.com/a/29882969

**Comment by CtrlC-Root** [comment]
So, if you're on a recent version of linux that has support for network namespaces you can work around it by creating a new network namespace that contains only the interface you want to use. Then you can run the `phonegap serve` command in that namespace. It's far from ideal, mostly because that means the interface isn't available in the default namespace, but it'll work in a VM or other similar situation.

```
# create a network namespace that only contains the public network interface
sudo ip netns add public
sudo ip link set eth1 netns public
sudo ip netns exec public dhclient

# create an alias to run commands using the public network namespace
echo 'alias run-public="sudo ip netns exec public"' >> /home/vagrant/.bashrc
```

You can now run phonegap using the `public` namespace.

```
run-public phonegap serve
```

**Comment by wearetheroyals** [comment]
Hey,

There is another way, slightly more hacky but it works :)

Goto into your Applications folder and right click PhoneGap and click Show Package Contents

open: Contents > Resources > app.nw > node_modules > connect-phonegap > lib > util > ip.js

You will see the following:

        // local address should be defined in the following fallback order:
        localAddress = localAddress ||
                       res.req.connection.localAddress ||
                       res.req.socket.localAddress ||
                       ip.address();
        localAddress = "192.168.178.157";
        callback(null, localAddress);

just above callback() I added localAddress = "192.168.178.157";

set that to your IP, restart the app and you're off.

**Comment by BrainSlugs83** [comment]
Why the hell is this closed? -- It's not fixed, and it's a major bug.

**Comment by hermwong** [comment]
This issue will be fixed with: #494

**Comment by mwbrooks** [comment]
*fingers crossed*

Thanks for all the comments on this issue - they helped guide the work!

**Comment by hermwong** [comment]
Multi-IP support was added with #494 and released in PhoneGap Desktop 0.2.1.
